apisix/plugins/cors.lua

```diff
@@ -133,7 +157,7 @@ local function set_cors_headers(conf, ctx)
     end
 
     core.response.set_header("Access-Control-Allow-Origin", ctx.cors_allow_origins)
-    if ctx.cors_allow_origins ~= "*" then
+    if ctx.cors_allow_origins ~= "*" or conf.allow_origins_by_regex ~= nil then
```

Comment:
I still don't understand why we need `or conf.allow_origins_by_regex ~= nil`.
If `allow_origins_by_regex` is given but not matched, it can't go into this function.
If `allow_origins_by_regex` is given but matched, the `cors_allow_origins` can't be `*` as `*` is just a mark used in `allow_origins`.

Reply:
if allow_origins_by_regex is matched, the  cors_allow_origins will be the matched domain. not `*`
for example. it we use `[".*.test.com"]` and the request is from a.test.com
then the cors_allow_origins will set to  `a.test.com`. not `*`

`conf.allow_origins_by_regex ~= nil` means we use regex to match many domains.
the Vary must set as `Origin `
f a request may contain a Access-Control-Allow-Origin with different values, then the CDN should always respond with Vary: Origin,
check 
https://stackoverflow.com/questions/25329405/why-isnt-vary-origin-response-set-on-a-cors-miss

Reply:
Yes. 

>if allow_origins_by_regex is matched, the cors_allow_origins will be the matched domain. not *

So `ctx.cors_allow_origins ~= "*"` will be true.
Therefore there is no need to add an additional condition.